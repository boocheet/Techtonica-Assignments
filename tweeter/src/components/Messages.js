import React from 'react'

export class Message {
  constructor(text, user) {
    let date = new Date()
    this.key = date.toString()
    this.text = text
    this.user = user
    this.date = date
  }
}

export function Messages({messages}) {
  //<Messages/>
  if (messages.length === 0) {
    return (
      <div>No posts found</div>
    );
  }
  return (
    <div>
    <h2>Posts:</h2>
    <ul>
      {messages.sort((a, b) => b.date - a.date).map((message, i) => (
        <li key="{i}">{message.text} - {message.user} - {message.key}</li>
      ))} 
    </ul>
    </div>
  );
}

// export default Messages