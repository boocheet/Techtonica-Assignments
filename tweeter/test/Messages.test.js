import React from 'react';
import Messages from '../src/components/Messages';
import { shallow, render, mount } from 'enzyme';


const message = 'Test Message';
let wrapped = shallow(<Messages>{message}</Messages>);

describe('Message', () => {
  it('should render the Message Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });
  it('renders the Messages children', () => { 
    expect(wrapped.find('ul').text()).toEqual(message);
  });
});
