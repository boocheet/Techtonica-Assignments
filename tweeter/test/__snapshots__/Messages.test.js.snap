// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render the Message Component correctly 1`] = `ShallowWrapper {}`;
